src/OpenSage.Game/Gui/Apt/ActionScript/ActionContext.cs

```diff
@@ -12,23 +12,39 @@ public sealed class ActionContext
         public ObjectContext Global { get; set; }
         public AptContext Apt { get; set; }
         public InstructionStream Stream { get; set; }
-        public Stack<Value> Stack { get; set; }
         public Value[] Registers { get; set; }
         public Dictionary<string, Value> Params { get; set; }
         public Dictionary<string, Value> Locals { get; set; }
         public bool Return { get; set; }
         public List<ConstantEntry> Constants { get; set; }
 
+        private Stack<Value> _stack;
```

Comment:
I made this private, so whenever a value is popped it also checks the registers & constants